### PR TITLE
feat: add typescript support

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

File: packages/searchbase/src/Results.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface Results {
+  // An array of results obtained from the applied query.
+  data: Array<Object>;
+
+  // Raw response returned by ES query
+  raw: Object;
+
+  // Results parser
+  parseResults: (results: Array<Object>) => Array<Object>;
+}
```
Comment (bietkul) [inline packages/searchbase/src/Results.d.ts]: A lot of properties are missing.

File: packages/searchbase/src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  DataField,
+  QueryFormat,
+  RequestStatus,
+  SortOption,
+  UpdateOn
+} from './types';
+import { Observable } from './observable';
+
+export class SearchBase {
+  static defaultQuery: (
```
Comment (bietkul) [inline packages/searchbase/src/index.d.ts]: constructor is missing.

File: packages/searchbase/src/types.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+type Suggestion = {
```
Comment (bietkul) [inline packages/searchbase/src/types.js]: What's the need of creating a separate file for types?
Types are more readable when they exist in the same file.

File: packages/searchbase/src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  sortByField: string;
+
+  dataField: string | Array<string | DataField>;
+
+  includeFields: Array<string>;
+
+  excludeFields: Array<string>;
+
+  sortOptions: Array<SortOption>;
+
```
Comment (bietkul) [inline packages/searchbase/src/index.d.ts]: I think you missed the `getter` properties.

File: packages/searchbase/src/types.js (hunk 3 above)
Comment (ShahAnuj2610) [reply packages/searchbase/src/types.js]: let's say we introduce `i18n`, it is always better to have all the strings in one place. Also, it becomes easier to search if all the `types` of the project lies at one place

File: packages/searchbase/src/index.d.ts [hunk 5]
```diff
@@ -151,4 +155,86 @@ export class SearchBase {
...
+    excludeFields,
+    transformQuery,
+    transformSuggestionsQuery,
+    transformRequest,
+    transformResponse,
+    beforeValueChange,
+    sortBy,
+    nestedField,
+    sortOptions
+  }: any);
```
Comment (bietkul) [inline packages/searchbase/src/index.d.ts]: Can you define the type of it instead using `any`.
All types will be the same as the `Searchbase` properties except `suggestions` & `results`.
suggestions => Array<Suggestions> // check the `Suggestion` type in `index.js`
results => Array<Object>

**Pull request merged**